### PR TITLE
Add Aggregation class.

### PR DESCRIPTION
**Pull request by dairyo**
To adapt thing-if AndroidSDK APIs, I added following 4 class:

* Aggregation: ([Android version](https://github.com/KiiPlatform/thing-if-AndroidSDK/blob/develop/thingif/src/main/java/com/kii/thingif/query/Aggregation.java))
* AggregatedResult: ([Android version](https://github.com/KiiPlatform/thing-if-AndroidSDK/blob/develop/thingif/src/main/java/com/kii/thingif/query/AggregatedResult.java))
* HistoryState: ([Android version](https://github.com/KiiPlatform/thing-if-AndroidSDK/blob/develop/thingif/src/main/java/com/kii/thingif/query/HistoryState.java))
* TimeRange: ([Android version](https://github.com/KiiPlatform/thing-if-AndroidSDK/blob/develop/thingif/src/main/java/com/kii/thingif/query/TimeRange.java))

I move 2 enumerations to internal enum of Aggregation. 

* FieldType
* Function as FunctionType

Related PR: #265

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/TimeRange.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//
+//  Created on 2017/01/30.
+//  Copyright (c) 2017 Kii. All rights reserved.
+//
+
+import Foundation
+
+open class TimeRange: NSCoding {
+
+    /** Start point of a time range. */
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/TimeRange.swift]: Please mention it is inclusive.

File: ThingIFSDK/ThingIFSDK/TimeRange.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    open let from: Date
+    /** End point of a time range. */
+    open let to: Date
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/TimeRange.swift:16]: please mentions it is inclusive

File: ThingIFSDK/ThingIFSDK/TimeRange.swift (hunk 1 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/TimeRange.swift]: I agree with mentioning inclusive or not is important but we should mention about it at using point such as `AggreagedResult.timeRange`. 

I think inclusive or exclusive depends on its use case.

What do you think?

Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/TimeRange.swift]: If we would like to provide more use cases, we need to provide these options for developer. Such as adding attribute to indicate inclusive or exclusive, left open or write open. 

But in our API, the timeRange is just used as inclusive case. We can provide more options, if we allow our developers for other options after server provide these possibilities.

**Comment by yongpingchen** [comment]
LGTM

**Pull request merged**